<?xml version="1.0" encoding="utf-8"?>
<launch>
  <arg name="robot_ns" default="qilin" />
  <arg name="launch_prefix" default="" /> set to value="gdbserver localhost:10000" for remote debugging
  <arg name="node_namespace" default="apriltag_ros_continuous_node" />
  <arg name="camera_name" default="/usb_cam" />
  <arg name="image_topic" default="image_rect" />
  <!-- USB camera -->
    <node name="usb_cam" pkg="usb_cam" type="usb_cam_node" output="screen" >
      <param name="video_device" value="/dev/usb_cam" />
      <param name="image_width" value="1280" />
      <param name="image_height" value="720" />
      <param name="pixel_format" value="mjpeg" />
      <param name="color_format" value="yuv420p" />
      <param name="framerate" value="30"/>
      <param name="camera_frame_id" value="usb_cam" />
      <param name="io_method" value="mmap"/>
      <param name="camera_info_url" value="file://$(find camera)/config/calibration_info.yaml" />
    </node>


    <!-- <node name="image_proc" pkg="image_proc" type="image_proc" output="screen"> -->
    <!--   <remap from="image_raw" to="/usb_cam/image_raw"/> -->
    <!--   <remap from="camera_info" to="/usb_cam/camera_info"/> -->
    <!--   <remap from="image_rect_color" to="/usb_cam/image_rect_color" /> -->
    <!-- </node> -->

    <!-- <include file="$(find apriltag_ros)/launch/continuous_detection.launch"> -->
    <!--   <remap from="camera_info" to="/usb_cam/camera_info" /> -->
    <!--   <arg name="camera_name" default="/usb_cam" /> -->
    <!--   <arg name="image_topic" default="image_rect_color" /> -->
    <!-- </include> -->
    <!-- <rosparam command="load" file="$(find camera)/config/tags.yaml" ns="/apriltag_ros_continuous_node"/> -->


<!--     <rosparam command="load" file="$(find apriltag_ros)/config/settings.yaml" ns="$(arg node_namespace)" /> -->
<!--     <rosparam command="load" file="$(find cooperation_landing)/config/apriltag/tags.yaml" ns="/apriltag_ros_continuous_node"/> -->

<!--     <node pkg="apriltag_ros" type="apriltag_ros_continuous_node" name="$(arg node_namespace)" clear_params="true" output="screen" launch-prefix="$(arg launch_prefix)" > -->
<!--     Remap topics from those used in code to those on the ROS network -->
<!--       <remap from="image_rect" to="/image_rect_color" /> -->
<!--       <remap from="camera_info" to="/usb_cam/camera_info" /> -->
<!--       <param name="publish_tag_detections_image" type="bool" value="true" />      default: false -->
<!--     </node> -->

    <!-- <node name="image_view" pkg="image_view" type="image_view" respawn="false" output="screen"> -->
    <!--   <remap from="image" to="/usb_cam/image_raw"/> -->
    <!--   <param name="autosize" value="true" /> -->
    <!-- </node> -->

</launch>
